src/main/scala/cromwell/engine/backend/jes/GcsAuth.scala

```diff
@@ -0,0 +1,71 @@
+package cromwell.engine.backend.jes
+
+import play.api.libs.json._
+
+object GcsAuth {
+  val boilerplateJson = Json.parse(s"""
+       |{
+       |  "auths" : {}
+       |}
+     """.stripMargin)
+
+  /**
+   * Add the JsObject parameters to the `auths` entry
+   */
+  def authTransformer(authJsons: Seq[JsObject]) = (__ \ 'auths).json.update {
+    __.read[JsObject] map { o =>
+      o ++ authJsons.reduce(_ ++ _)
+    }
+  }
+
+  /**
+   * Generates a json containing auth information based on the parameters provided.
+   * @return a string representation of the json
+   */
+  def generateJson(dockerAuth: Option[DockerAuthInformation], userAuth: Option[GcsUserAuthInformation]) = {
+    Seq(dockerAuth, userAuth).flatten map { _.toJsObject } match {
```

Comment:
I find flatten better here because we have `Option`s in the `Seq` before even mapping the values, and we only want to map the ones that are not `None`. Also with this approach the values mapped are not `Option`s but actual values, which avoid having subsequent function dealing with `Option` parameter etc...